### ISSUE TITLE
ApiServer spawns an unbounded thread per connection with no concurrency cap

### ISSUE THREAD

**Issue by tylerreckart** [tylerreckart/arbiter#49]
## Summary

`ApiServer::accept_loop()` (`src/api_server.cpp:9756-9782`) spawns a new detached `std::thread` for every accepted connection with no upper bound:

```cpp
void ApiServer::accept_loop() {
    while (running_.load()) {
        int client = ::accept(listen_fd_, nullptr, nullptr);
        ...
        active_connections_.fetch_add(1, std::memory_order_acq_rel);
        std::thread([this, client]() {
            try { handle_connection(client); }
            catch (...) { /* drop */ }
            ::close(client);
            if (active_connections_.fetch_sub(1, std::memory_order_acq_rel) == 1) {
                std::lock_guard<std::mutex> lk(drain_mu_);
                drain_cv_.notify_all();
            }
        }).detach();
    }
}
```

`active_connections_` is only consulted for graceful-drain bookkeeping on shutdown (waiting for in-flight connections to finish) — it's never checked against a limit before spawning a new thread.

## Impact

A burst of connections (accidental client retry storm, or a deliberate flood) creates one OS thread per connection with no ceiling. Each thread carries its own stack (default ~8MB on many platforms unless overridden) plus scheduler overhead; enough concurrent connections exhausts memory or hits the OS thread-count limit, at which point `std::thread` construction throws and the accept loop's `catch` (if any at that level — worth checking) determines whether the whole server goes down or just that connection.

## Suggested fix

Add a concurrency cap: track `active_connections_` against a configurable maximum and either block/reject-with-503 or queue new accepts once the cap is hit, rather than spawning unconditionally. A fixed-size thread pool (or at least a semaphore gating thread creation) would bound worst-case memory/thread usage under load.
